coc/client.py

```diff
@@ -1066,37 +1066,29 @@ async def get_current_war(
                 raise PrivateWarLog(exception.response, exception.reason) from exception
             return get_war
 
-        is_prep = league_group.state == "preparation"
-
-        if cwl_round is WarRound.current_war and league_group.state == "preparation":
-            return None  # for round 1 and 15min prep between rounds this is a shortcut.
-        elif cwl_round is WarRound.current_preparation and league_group.state == "ended":
-            return None  # for the end of CWL there's no next prep day.
-        elif cwl_round is WarRound.previous_war and len(league_group.rounds) == 1:
-            return None  # no previous war for first rounds.
-        elif cwl_round is WarRound.current_war and league_group.state == "ended":
-            round_tags = league_group.rounds[-1] # for the end of CWL current_war should give the last war
-        elif cwl_round is WarRound.previous_war and is_prep:
-            round_tags = league_group.rounds[-2]
-        elif cwl_round is WarRound.previous_war:
-            round_tags = league_group.rounds[-3]
-        elif cwl_round is WarRound.current_war:
-            round_tags = league_group.rounds[-2]
-        elif cwl_round is WarRound.current_preparation:
-            round_tags = league_group.rounds[-1]
-        else:
-            return None
+        if league_group.state == "notInWar":
+            return get_war
 
         kwargs["league_group"] = league_group
         kwargs["clan_tag"] = clan_tag
-        async for war in self.get_league_wars(round_tags, cls=cls, **kwargs):
-            if war.clan_tag == clan_tag:
-                return war
-            elif war.opponent.tag == clan_tag:
-                tmp = war.clan
-                war.clan = war.opponent
-                war.opponent = tmp
-                return war
+        war = None
+        all_wars = []
+        async for w in league_group.get_wars_for_clan(clan_tag):
+            all_wars.append(w)
+            if WarRound.current_war:
+                if str(w.state) == "warEnded":
+                    war = w
+                if str(w.state) == "inWar":
+                    war = w
+            elif WarRound.current_preparation:
+                if str(w.state) == "preparation":
+                    war = w
+        if WarRound.current_war and war is None:
+            try:
+                war = all_wars[0]
+            except:
+                pass
+        return war
```

Comment:
All of this has nothing to do with the raid events right? Then please make a seperate PR for it.